File: yak-gl-util/src/main/java/net/yakclient/opengl/util/VacantTexture.java
package net.yakclient.opengl.util;

public class VacantTexture implements YakTexture {
    @Override
    public boolean hasAlpha() {
         return false;
    }

    @Override
    public void bind() {
        //Nothing to do
    }

    @Override
    public int getImageHeight() {
        return 0;
    }

    @Override
    public int getImageWidth() {
        return 0;
    }

    @Override
    public float getHeight() {
        return 0;
    }

    @Override
    public float getWidth() {
        return 0;
    }

    @Override
    public int getTextureHeight() {
        return 0;
    }

    @Override
    public int getTextureWidth() {
        return 0;
    }

    @Override
    public void release() {
        //Nothing to do
    }

    @Override
    public TexAggregation generateTexs(VerticeAggregation vertices) {
        return new TexAggregation();
    }

}

File: util/src/main/kotlin/net/yakclient/graphics/util/VacantTexture.kt
package net.yakclient.graphics.util

public class VacantTexture : Texture {
    override val parent: Texture? = null
    override val height: Int = 0
    override val width: Int = 0
    override val offsetX: Int = 0
    override val offsetY: Int = 0

    override fun subTexture(x: Int, y: Int, width: Int, height: Int): Texture = this

    override fun bind() {}

    override fun release() {}
}